File: lab2/cmd/main.go
package main

import (
	u "CryptoLabs/lab2/internal/app/utils"
	"flag"
)

var (
	key        string
	mode       string
	initVector string
	file       string
	encryption bool
	decryption bool
	debug      bool
)

func init() {
	flag.StringVar(&key, "k", "", "key in hex")
	flag.StringVar(&mode, "m", "", "mode (ecb or cbc)")
	flag.StringVar(&initVector, "i", "", "key in hex")
	flag.StringVar(&file, "f", "", "file with text")
	flag.BoolVar(&encryption, "e", true, "flag for operation in encryption mode")
	flag.BoolVar(&decryption, "d", false, "flag for operation in decryption mode")
	flag.BoolVar(&debug, "g", false, "flag for operation in debug mode")
}
func main() {
	flag.Parse()
	u.Menu(key, mode, initVector, file, encryption, decryption, debug)
}

File: lab2/main.py
import argparse
import cbc
import ecb
from constants import BLOCK_SIZE

parser = argparse.ArgumentParser(prog='cipher', description='AES encryption util')
parser.add_argument('--version', action='version', version='%(prog)s 1.0',
                    help='Project version')
parser.add_argument('-m', '--mode', required=True,
                    help='Encryption or Decryption algorithm')
group = parser.add_mutually_exclusive_group(required=True)
group.add_argument('-e', '--enc', action='store_true', help='Encryption mode')
group.add_argument('-d', '--dec', action='store_true', help='Decryption mode')
parser.add_argument('-k', '--key', required=True, help='Key in hex')
parser.add_argument('-i', '--iv',
                    help='IV for cbc mode')
parser.add_argument('file',
                    help='File with message')

args = parser.parse_args()
mode = args.mode
file_path = args.file
key = args.key
iv = args.iv
result = None

if len(bytearray.fromhex(key)) != BLOCK_SIZE:
    raise Exception('Invalid size of key ...')

if mode == 'cbc' and iv is None:
    raise Exception('Cannot find iv in cbc mode ...')

if mode == 'cbc' and (len(iv) != BLOCK_SIZE):
    raise Exception('Invalid size of iv vector')

with open(file_path, 'r') as input_file:
    text = input_file.readline().rstrip('\n')
    if args.enc:
        if mode == 'cbc':
            result = cbc.cbc_encryption(text, bytearray.fromhex(key), bytearray.fromhex(iv))
        elif mode == 'ecb':
            result = ecb.ecb_encryption(text, bytearray.fromhex(key))
    elif args.dec:
        if mode == 'cbc':
            result = cbc.cbc_decryption(text, bytearray.fromhex(key), bytearray.fromhex(iv))
        elif mode == 'ecb':
            result = ecb.ecb_decryption(text, bytearray.fromhex(key))
    print("Done ...")
    print(result)
